src/kernels/raw/finder/jupyterPaths.node.ts

```diff
@@ -125,9 +128,9 @@ export class JupyterPaths {
      * https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernel-specs
      */
     @traceDecoratorVerbose('Get Kernelspec root path')
-    public async getKernelSpecRootPaths(cancelToken?: CancellationToken): Promise<string[]> {
+    public async getKernelSpecRootPaths(cancelToken?: CancellationToken): Promise<Uri[]> {
         // Paths specified in JUPYTER_PATH are supposed to come first in searching
-        const paths = new Set<string>(await this.getJupyterPathPaths(cancelToken));
+        const paths = new Set<Uri>(await this.getJupyterPathPaths(cancelToken));
```

Comment:
I don't think this would work, Uri.file('a') !== Uri.file('a')
Hence the uniqueness isn't going to work with the Set class

Reply:
Thanks forgot about this one. I need to use the comparison key.